client/my-sites/site-indicator/site-indicator.jsx

```diff
@@ -295,4 +357,32 @@ export default React.createClass( {
 			</div>
 		);
 	}
-} );
+
+	render() {
+		const { site, siteIsJetpack } = this.props;
+
+		return (
+			<div className="site-indicator__wrapper">
+				{ siteIsJetpack && <QuerySiteConnectionStatus siteId={ site.ID } /> }
```

Comment:
It's the same reasoning as why we allow siteId to be falsey, so we don't need to do:
```jsx
{ siteId && siteIsJetpack && <QuerySiteConnectionStatus siteId={ siteId } /> }
```

I can go either way on this one since a bare `<QuerySiteConnectionStatus siteId={ siteId } /> works too, it'll just fail for wpcom sites.